### PR TITLE
feat: better UX on slow connections

### LINKED ISSUE #126: Cache and hide asset loading

### PR DESCRIPTION
**Pull request by belohlavek**
kapture 2019-02-19 at 18 16 18

☝️ Tested on slow connection 

- Added loader before before react mounts
- Updated ECS (now uses IndexedDB cache for 3d assets)
- Updated spinner on sidebar 
- Updated placeholder graphic when asset thumbnails fail to load
- Preview is reused and some initialization steps are now skipped

Also on the content server's side assets are now served with a better cache policy

Closes #126

### CODE REVIEW

**Comment by nicosantangelo** [comment]
I don't know if this is intentional, but loading an editor URL directly shows me the loader first and then the loading spinner for quite some time. Having to see both it's a bit unnerving and feels ( _to me_ ) like the loading more time.

loading

☝️ the gif is using my normal connection

**Comment by nicosantangelo** [comment]
Same goes for loading the homepage, loading without cache on a good connection in master:

before-loading

and with the loader:

after-loading


----

I know this is aimed to make the experience of users with a bad connection better, but my point is that I wouldn't want to degrade the "normal" experience (if possible ¯\_(ツ)_/¯)

File: src/components/TemplateCard/TemplateCard.css [hunk 1]
```diff
@@ -14,6 +15,7 @@
   overflow: hidden;
   white-space: nowrap;
   text-overflow: ellipsis;
+  user-select: none;
```
[the remark this answers is not in the document]
Comment (belohlavek) [reply src/components/TemplateCard/TemplateCard.css]: Yep, you may want to copy the project name

File: src/components/AssetCard/VerticalCard/VerticalCard.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ export default class VerticalCard extends React.PureComponent<Props> {
 
     return (
       <div className={classes} title={name}>
-        <img className="thumbnail" src={thumbnail} />
+        <img className="thumbnail" src={thumbnail} alt="" />
```
[the remark this answers is not in the document]
Comment (belohlavek) [reply src/components/AssetCard/VerticalCard/VerticalCard.tsx]: We need to hide text and the broken file icon, that's why it is empty :)

**Comment by belohlavek** [comment]
kapture 2019-02-20 at 10 03 00
Fast connection with/without cache

kapture 2019-02-20 at 10 04 06
Slow connection with/without cache

So as long as we keep the time-to-mount under 2s we should be ok and you should never see the loader. I think 2s is plenty anyways.

**Comment by nicosantangelo** [comment]
**Much** better! LGTM

**Pull request merged**